mwclient/entity.py

```diff
@@ -0,0 +1,271 @@
+        super(Item, self).setinfofromwbgetentities(result)
+        self._sitelinks = dict()
+        for wiki in result['sitelinks']:
+            site = result['sitelinks'][wiki]['site']
+            title = result['sitelinks'][wiki]['title']
+            badges = result['sitelinks'][wiki]['badges']
+            self._sitelinks[site] = {'title': title, 'badges': badges}
+
+    @property
+    def sitelinks(self):
+        """Sitelinks dictionnary with title, and badges per site.
+
+        Example:
+            >>> import mwclient
+            >>> site = mwclient.WikiBaseSite(('https', 'www.wikidata.org'))
+            >>> q = mwclient.entity.Item(site, 'Q3340172')
+            >>> q.sitelinks
+        """
+        if self._sitelinks is None:
+            entities = self.site.api('wbgetentities', ids=self.entity)
+            result = entities['entities'][self.entity]
+            self.setinfofromwbgetentities(result)
+        return self._sitelinks
+
+    def __repr__(self):
+        """Item representation."""
+        return "<Item object %s (%s)>" % (self.entity, self.site.host)
+
+
+class Property(Entity):
+
+    """Wikibase Property."""
+
+    def __init__(self, site, entity):
+        """Constructor.
+
+        Args:
+            site (WikiBaseSite): reference to a WikiBaseSite
+            entity (str): Q number of the entity.
+        """
+        # Normalizing entity name
+        super(Property, self).__init__(site, 'P' + entity.upper().lstrip('P'))
+
+    def __repr__(self):
+        """Property representation."""
+        return "<Property object %s (%s)>" % (self.entity, self.site.host)
+
+
+class Claim(object):
+
+    """Claim
+
+    Attributes:
+        prop (str): property id.
+        snak (dict): snak with all values return in mainsnak from API call.
+        datatype (str): datatype ('wikibase-item', 'string', etc.)
+        string (str): string value, None if not 'string' datatype
+        item (Item): Item value, None if not 'wikibase-item' datatype
+        property_value (Property): Property, None if not 'wikibase-property'
+            datatype
+        latitude (float): latitude, None if not 'globe-coordinate' datatype
+        longitude (float): longitude, None if not 'globe-coordinate' datatype
+        quantity (dict): quantity dictionnary, None if not 'quantity' datatype
+    """
+
+    def __init__(self, site, prop, datatype, value, snak=None):
+        """Constructor"""
+        self.site = site
+        self.prop = prop
+        self.datatype = datatype
+        self.value = value
+        self.snak = snak
+
+    @classmethod
+    def fromsnak(cls, site, snak):
+        """Claim from snak dictionnary.
+
+        Args:
+            site (mwclient.WikiBaseSite): site
+            snak (dict): snak dictionnary
+        """
+        return cls(site, snak['property'],
+                   snak['datatype'],
+                   snak['datavalue']['value'])
```

Comment:
Cool! I guess `snaktype` could be stored on the Claim object? Not sure what's best to do with `Claim.value`.. perhaps just set it to None?